File: LABA5/parserPack/DoubleParser.java
package parserPack;

import exceptionsPack.NullFieldException;
import exceptionsPack.NumberOutOfBoundsException;

public class DoubleParser {
    public static final int UPPER_BOUND = 274;
    public static Double parse(String line) throws NullFieldException, NumberOutOfBoundsException {
        Double n = (Double)(double)(float)FloatParser.parse(line);
        if (n > UPPER_BOUND) throw new NumberOutOfBoundsException();
        return n;
    }
}

File: LABA5kt/src/parserPack/DoubleParser.kt
package parserPack

import exceptionsPack.NullFieldException
import exceptionsPack.NumberOutOfBoundsException

object DoubleParser {
    const val UPPER_BOUND: Int = 274

    /**
     * Parses the input string and returns a Double value, while handling specific exceptions.
     *
     * @param  line  the input string to be parsed
     * @return       the parsed Double value
     */
    @Throws(NullFieldException::class, NumberOutOfBoundsException::class)
    fun parse(line: String?): Double {
        val n = FloatParser.parse(line!!).toDouble()
        if (n > UPPER_BOUND) throw NumberOutOfBoundsException()
        return n
    }
}